lib/coap/connection.ex

```diff
@@ -177,6 +177,7 @@ defmodule CoAP.Connection do
 
     Wrap the adapter and the client in a handler
   """
+  @impl GenServer
   def init([server, {adapter, endpoint}, {ip, port, token} = _peer, config]) do
     {:ok, handler} = start_handler(adapter, endpoint)
```

Comment:
After calling `start_handler` (or in `start_handler`) should we link the handler to connection https://hexdocs.pm/elixir/Process.html#link/1 so that an exit in one results in an exit in the other. Then we would not need to send `:connection_end` to the handler.

Reply:
In the case processes are linked:
* when Connection stops, Handler will stop -> ok
* when Handler stops, Connection will stop -> not ok

Maybe monitoring Connection from Handler process (`Process.monitor/1`) ?

Reply:
latest commit: use `Process.monitor` instead of custom message